ql/cashflows/cpicoupon.cpp

```diff
@@ -42,14 +43,86 @@ namespace QuantLib {
+    : CPICoupon(baseCPI,
+                Null<Date>(),
+                paymentDate,
+                nominal,
+                startDate,
+                endDate,
+                zeroIndex,
+                observationLag,
+                observationInterpolation,
+                dayCounter,
+                fixedRate,
+                spread,
+                refPeriodStart,
+                refPeriodEnd,
+                exCouponDate) {}
+
+    CPICoupon::CPICoupon(const Date& baseDate,
+                         const Date& paymentDate,
+                         Real nominal,
+                         const Date& startDate,
+                         const Date& endDate,
+                         const ext::shared_ptr<ZeroInflationIndex>& zeroIndex,
+                         const Period& observationLag,
+                         CPI::InterpolationType observationInterpolation,
+                         const DayCounter& dayCounter,
+                         Real fixedRate,
+                         Spread spread,
+                         const Date& refPeriodStart,
+                         const Date& refPeriodEnd,
+                         const Date& exCouponDate)
+    : CPICoupon(Null<Real>(),
+                baseDate,
+                paymentDate,
+                nominal,
+                startDate,
+                endDate,
+                zeroIndex,
+                observationLag,
+                observationInterpolation,
+                dayCounter,
+                fixedRate,
+                spread,
+                refPeriodStart,
+                refPeriodEnd,
+                exCouponDate) {}
+
+    CPICoupon::CPICoupon(Real baseCPI,
+                         const Date& baseDate,
+                         const Date& paymentDate,
+                         Real nominal,
+                         const Date& startDate,
+                         const Date& endDate,
+                         const ext::shared_ptr<ZeroInflationIndex>& zeroIndex,
+                         const Period& observationLag,
+                         CPI::InterpolationType observationInterpolation,
+                         const DayCounter& dayCounter,
+                         Real fixedRate,
+                         Spread spread,
+                         const Date& refPeriodStart,
+                         const Date& refPeriodEnd,
+                         const Date& exCouponDate)
+    : InflationCoupon(paymentDate,
+                      nominal,
+                      startDate,
+                      endDate,
+                      0,
+                      zeroIndex,
+                      observationLag,
+                      dayCounter,
+                      refPeriodStart,
+                      refPeriodEnd,
+                      exCouponDate),
       baseCPI_(baseCPI), fixedRate_(fixedRate), spread_(spread),
       observationInterpolation_(observationInterpolation) {
```

Comment:
It looks like we're not saving `baseDate` here, am I mistaken?

Reply:
Oh yes, thanks for spotting this. I will update it